### PR TITLE
Edit for clarity

### PR DESCRIPTION
**Pull request by tvalentyn**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by tvalentyn** [comment]
R: @AnandInguva

File: website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md [hunk 1]
```diff
@@ -224,8 +224,9 @@ Use a constraint file to ensure that Beam dependencies in the launch environment
 
 The launch environment translates the  pipeline graph into a [runner-independent representation](https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto). This process involves serializing (or pickling) the code of the transforms. The serialized content is deserialized on the workers. If the runtime worker environment significantly differs from the launch environment, runtime errors might occur for the following reasons:
 
+* The Apache Beam version and the Python major.minor versions must match in the submission and runtime environments. Otherwise, the pipeline might fail with errors like `Pipeline construction environment and pipeline runtime environment are not compatible`. On older SDK versions, the error might be reported as `SystemError: unknown opcode`.
```
Comment (AnandInguva) [inline website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md:227]: the wording when I first read feels like `apache beam version` == `python versions`

Comment (tvalentyn) [reply website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md]: PTAL

File: website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md [hunk 2]
```diff
@@ -224,8 +224,9 @@ Use a constraint file to ensure that Beam dependencies in the launch environment
 
 The launch environment translates the  pipeline graph into a [runner-independent representation](https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto). This process involves serializing (or pickling) the code of the transforms. The serialized content is deserialized on the workers. If the runtime worker environment significantly differs from the launch environment, runtime errors might occur for the following reasons:
 
+* The Apache Beam version  must match in the submission and runtime environments. Python major.minor versions must match as well. Otherwise, the pipeline might fail with errors like `Pipeline construction environment and pipeline runtime environment are not compatible`. On older SDK versions, the error might be reported as `SystemError: unknown opcode`.
```
Comment (AnandInguva) [inline website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md:227]: IIRC, if python versions are not same, it used to fail with dill errors. Is this the one dill output as error?

**Review by AnandInguva — APPROVED** [review]

File: website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md (hunk 2 above)
Comment (tvalentyn) [reply website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md:227]: we now have a better error, https://github.com/apache/beam/blob/5bf6fbf26708e06cc7b2fe286e400194890ffa49/sdks/python/apache_beam/runners/worker/bundle_processor.py#L851

**Pull request merged**